dummy_telemetry/dummy_telemetry.ino

```diff
@@ -8,68 +8,134 @@ Assumptions :
+  Data is sent by "line" ie. by packet of data ending with a newline (line feed '\n').
   To check the integrity of the data sent (at least it's completeness), the first character is :
-    - "H" for a header line
-    - "N" for any other line
-  and the last character (before the newline) is always "E".
+    - "@" for a header line
+    - "#" for a data line
+    - "%" for a calibration line
+    - "$" for a message line
   The first character is used to check if we received the beginning of the line. The last character is used to check
   if the line is complete (ie if the serial read did not timeout before the end of the line)
 
   Before transmitting values, the telementry receiver sends a recognizable BONJOUR string so that the reader can
   identify what device is sending data
 
-  The transmission starts by the sending of the data header line to the ground station. This header is sent once.
-  Then the data is sent
-
 Comments :
   - We should send the time of the data sampling along the samples
   - The link could be bi-directionnal while the rocket is on the launch pad
   - We need to define a nomenclature between the OBC and the Dashboard to know how to process the data
 
 */
 
-#define NEWLINE "N"
-#define NEWHEAD "H"
-#define ENDLINE "E"
-#define SEPDATA "\t"
+#define START_HEAD "@"
+#define START_DATA "#"
+#define START_CALI "%"
+#define START_MESS "$"
+#define SEP_DATA "&"
+#define SEP_CALI ":"
+#define END_LINE "\n"
+
 #define BONJOUR "TELEMETRY"
 
 // Change this to use a different baudrate
 #define BAUDRATE 115200
 
-
 int cpt = 0;
 unsigned long time;
 
-
-void setup() {
+void setup()
+{
   Serial.begin(BAUDRATE);
-  // Send string to indicate that this device is the telemetry receiver
-  Serial.println(BONJOUR);
-  // Send the header to the ground station
-  Serial.print(NEWHEAD);
-  Serial.print("Millis");Serial.print(SEPDATA);
-  Serial.print("Data");
-  Serial.println(ENDLINE);
 
-  delay(1000);
+  // This is sent by the telemetry receiver
+  bonjour();
+  // Everything after this is a simulation of what is received by the telementry receiver
+  // and forwarded to the Dashboard
+  
+  health_check();
+
+  delay(100);
 }
 
 // Dummy code to send arbitrary data to the ground station
 // The data should match the header
-void loop() {
+void loop()
+{
   if (cpt >= 50)
-  { 
+  {
     cpt = 0;
   }
   time = millis();
-  Serial.print(NEWLINE);
-  Serial.print(time);Serial.print(SEPDATA);
+  Serial.print(START_DATA);
+  Serial.print(time);
+  Serial.print(SEP_DATA);
   Serial.print(cpt + 1);
-  Serial.println(ENDLINE);
+  Serial.print(END_LINE);
   cpt++;
 
   delay(10);
+}
 
+void bonjour()
+{
+  // Send string to indicate that this device is the telemetry receiver
+  Serial.print(BONJOUR);Serial.print(END_LINE);
+}
+
+void health_check()
+{
+  Serial.print(START_MESS);
+  Serial.print("Health check started");
+  Serial.print(END_LINE);

```

Comment:
How does the Serial.print and buffer in the computer work together? Like, can the buffer just read the 'START_MESS' and message but not the 'END_LINE' or just only the 'START_MESS' and not the rest? Probably not a problem but otherwise it will miss a whole message during flight for example, the probability should be very low if it read a multiple of 3 each time, or how does it work? Or is the delay(10) enough for dashboard.py to read the whole segment?

Reply:
Everything that is received by the computer is stored in the buffer. Each time the python code asks for a line the oldest bytes received are fetched until the endline character is found so we should not miss any received data. I have no idea if we can empty the buffer faster than it is filled with new data but I guess yes. And the 10ms delay is just to simulate the latency of the STM. We should do a proper speedtest soon to check the maximum throughput possible but maybe not before merging this

Reply:
Okey, see what you mean, but I was wondering how the receiving of data and when the reading of the buffer happens, like if the buffer could be read during an incoming message so that just half the message is read and not the whole things with the end symbol.

Reply:
For now we use `readline()` to read the buffer so it waits until the end line character is received to end the reading. No data is lost except if we get a timeout but that should not occur under normal operations